src/api/sync.rs

```diff
@@ -501,12 +507,69 @@ impl ApiRepo{
     /// api.model("gpt2".to_string()).info();
     /// ```
     pub fn info(&self) -> Result<RepoInfo, ApiError> {
-        let url = format!("{}/api/{}", self.api.endpoint, self.repo.api_url());
-        let response = self.api.client.get(&url).call().map_err(Box::new)?;
+        Ok(serde_json::from_value(self.detailed_info().send()?)?)
+    }
+
+    /// Get a builder to help create a request to the hub API
+    /// ```
+    /// # use hf_hub::api::tokio::Api;
+    /// # tokio_test::block_on(async {
+    /// let api = Api::new().unwrap();
+    /// api.model("gpt2".to_owned())
+    ///     .detailed_info()
+    ///     .query("blobs".to_owned(), "true".to_owned())
+    ///     .send();
+    /// # })
+    /// ```
+    pub fn detailed_info(&self) -> DetailedInfoBuilder {
+        DetailedInfoBuilder::new(self.api.clone(), &self.api.endpoint, &self.repo)
+    }
+}
 
-        let model_info = response.into_json()?;
+/// Builder to help create a request to the hub API
+/// in order to get detailed info about the repository
+pub struct DetailedInfoBuilder {
+    api: Api,
+    query_params: Option<Vec<(String, String)>>,
+    url: String,
+}
 
-        Ok(model_info)
+impl DetailedInfoBuilder {
+    fn new(api: Api, endpoint: &str, repo: &Repo) -> Self {
+        let url = format!("{}/api/{}", endpoint, repo.api_url());
+        Self {
+            api,
+            query_params: None,
+            url,
+        }
+    }
+}
+
+impl DetailedInfoBuilder {
+    /// Add a query parameter to the request
+    pub fn query(mut self, key: String, value: String) -> Self {
+        if let Some(mut qps) = self.query_params {
+            qps.push((key, value));
+            self.query_params = Some(qps);
+        } else {
+            self.query_params = Some(vec![(key, value)]);
+        }
+        self
+    }
+
+    /// Send the request and get the resulting JSON
+    pub fn send(self) -> Result<Value, ApiError> {
+        let mut request = self.api.client.get(&self.url);
+        if let Some(qps) = self.query_params {
+            let qps: Vec<(&str, &str)> = qps
+                .iter()
+                .map(|qp| (qp.0.as_ref(), qp.1.as_ref()))
+                .collect();
+            request = request.query_pairs(qps);
+        }
+        let response = request.call().map_err(Box::new)?;
+
+        Ok(response.into_json()?)
```

Comment:
```suggestion
impl DetailedInfoBuilder {
    fn new(api: &ApiRepo) -> Self {
        let url = format!("{}/api/{}", endpoint, repo.api_url());
         let mut request = self.api.client.get(&url);
        Self {
           request
        }
    }
}

impl DetailedInfoBuilder {
    /// Add a query parameter to the request
    pub fn query(mut self, key: String, value: String) -> Self {
        self.request = self.request.query(key, value)
        self
    }

    /// Send the request and get the resulting JSON
    pub fn send(self) -> Result<Value, ApiError> {
        let response = self.request.call().map_err(Box::new)?;
        Ok(response.into_json()?)
```

WDYT ?

Reply:
Maybe even Scratch the builder, and return the `Request` directly.. One less struct and full control over the querying to the user...

Reply:
Then the fn could be rename `info_request` maybe ?

Reply:
Yep agree, `info_request` sounds good

Reply:
looks better now !